enhancements/operator-hub-check.md

```diff
@@ -0,0 +1,215 @@
+
+The error raised will be accordingly with the Kubernetes version that we have, see:
+
+| Conditional |Type |
+| ------ | ----- |
+| operator bundle supports k8s => `1.22` and do not uses `apiextensions/v1` ÒR has [webhookdefinitions](https://github.com/operator-framework/operator-sdk/blob/v1.4.2/testdata/go/v2/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml#L203-L205) and do not have the API `v1` listed | error |
+| operator bundle supports k8s `=> 1.16 <= 1.22` and uses `apiextensions/v1beta1` ÒR `admissionregistration.k8s.io/v1beta1` | warning |
+
+**Why we need to check if the api v1 is not used instead of only check if it has any CRD or Webhook with the deprecated `v1beta1`?**
+		
+The operator author might provide a solution that works for both scenarios which means that will work when installed in the previous version of `Kubernetes < 1.16` and also in the upper versions `=> 1.22`. However, for the operator works well in any Kubernetes version `>=1.22`, we know that it is mandatory it has the latest APIs versions `v1`. See [here](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.0.0-beta.0/test/e2e/v3/plugin_cluster_test.go#L99-L101), for example, that in the e2e test for Kubebuilder, we check what is the cluster version to apply and test the API versions accordingly. And then, note that we can find both APIs in the webhooks fields of the CSV as well such as:
+
+```
+  webhookdefinitions:
+  - admissionReviewVersions:
+    - v1beta1
+    - v1   
+``` 
+
+**IMPORTANT** 
+
+> Kubernetes version `1.22` was not released until now which means that the deprecated APIs still supported for all clusters. Also, at the same way none OpenShift version which is using Kubernetes API `1.22` was released. Then, we cannot raise errors until it happens for the scenarios where we are `minKubeVersion` is empty and we are assuming all possible versions.
+> However, for the checks where the `--optional-values="k8s=1.22"` was used we know that the goal is to verify if the operator bundle will work òn `1.22` then, we can already raise the `error` result. 
+
+Then, it means that:
+- `--optional-values="kubeVersion=value"` flag with a value `=> 1.16 <= 1.22` the validator will return result as warning.
+- `--optional-values="kubeVersion=value"` flag with a value `=> 1.22` the validator will return result as error.
+- `minKubeVersion >= 1.22` return the error result.
+
+**Currently (Before `1.22` release)**
+- `minKubeVersion` empty/nil return the warning result.
+
+**After `1.22` k8s release**
+- `minKubeVersion >= 1.22 or empty/nil`  return the error result.
+
```

Comment:
I need to add here that the validators will need to be able to receive a string map and check if it contains any valid key for the specific check.